go/parser.go

```diff
@@ -13,6 +13,7 @@ const OPERATOR = "operator"
 type Evaluatable interface {
 	Evaluate(variables []string) bool
 	ToString() string
+	IsBinaryOperator() bool
```

Comment:
I see that you've used two different solutions. You are either type checking against the struct/class/ect type. Or you've added an interface. Unfortunately I think this method now becomes part of the public API. Is there a way to avoid this?

Reply:
It might be as simple as using `isBinaryOperator` instead.

Reply:
I know (but I did not add an interface that I am aware of; I just extended the existing interface).
But it was much simpler than to dive into these languages and find out if they provide an `is-instance-of(Class)` mechanism and how it works.

* Come up with a better approach that uses `is-instance-of(Class)` in Go, Perl, … and I am happy to change it
* Provide me with a better solution that does not expose the method `isBinaryOperator()` to an out-of-module user and allows to use this method within the module.

Reply:
Not tested.

Go: You can test if `x` is an instance of `y` with `x.(y)`

https://stackoverflow.com/a/72195588/3945473

Perl: You can use `isa`.

https://stackoverflow.com/a/1731655/3945473

Reply:
Thanks for the info.